src/character/complete.rs

```diff
@@ -9,9 +9,21 @@ use traits::{AsChar, AtEof, FindToken, InputIter, InputLength, InputTakeAtPositi
 use traits::{Compare, CompareResult};
 use error::ErrorKind;
 
-/// recognizes one character
+/// Recognizes one character.
 ///
-/// *complete version*: will return an error if there's not enough input data
+/// *complete version*: Will return an error if there's not enough input data.
+///
+/// # Example
+///
+/// ```
+/// # use nom::{Err, error::ErrorKind};
+/// # use nom::character::complete::char;
+/// # fn main() {
+/// assert_eq!(char::<_, (&str, ErrorKind)>('a')("abc"), Ok(("bc", 'a')));
```

Comment:
It's not a good idea to write `char::<_, (&str, ErrorKind)>` everywhere, it makes the example hard to read. If type inference on the error type is an issue, it should be better to write:
```rust
fn parser(input: &str) -> IResult<&str, &str> {
  char('a')
}
```